flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -123,6 +123,12 @@
 
 	private String jobName = DEFAULT_JOB_NAME;
 
+	/**
+	 * If there are some stream edges that can not be chained and the shuffle mode of edge is not specified,
+	 * translate these edges into `BLOCKING` result partition type.
+	 */
+	private boolean blockingAfterChainingOff = false;
```

Comment:
Why add this property?  Planner can get `StreamGraph` and set the property to it.

Reply:
Yes, planner could set `StreamGraph` directly. But that's a work-around way for 1.9. Actually the `StreamGraphGenerator` should be the only entry of `StreamGraph`. So just keep it here.